ci/test_conda.sh

```diff
@@ -4,14 +4,6 @@
 
 set -euo pipefail
 
-DISTRO=`cat /etc/os-release | grep "^ID=" | awk 'BEGIN {FS="="} { print $2 }'`
-
-if [ "$DISTRO" = "ubuntu" ]; then
-  apt-get update
-  apt remove --purge `dpkg --get-selections | grep cuda-nvvm | awk '{print $1}'` -y
-  apt remove --purge `dpkg --get-selections | grep cuda-nvrtc | awk '{print $1}'` -y
-fi
```

Comment:
This is not related to the other changes, but it was causing problems in CI, and isn't necessary anymore since we're getting all dependencies from `pixi` in this script, so I removed it.

Reply:
I think the reason this remained is that the CUDA pathfinder can mistakenly find system libraries ahead of ones in the environment, so we remove them to make sure we're testing with the right libraries. This was discussed in https://github.com/NVIDIA/cuda-python/issues/839 - note that although the issue is closed, I believe the fix only makes it less likely to find the library from the system install of the toolkit, not impossible.

Reply:
The behavior now should be correct. pip wheels will be found first, then conda packages, then a default system search using `dlopen` which has its own hierarchy.

As far as I know the only issue that remains is that if you're trying to find system libraries (as opposed to pip wheels or conda packages) that nvvm doesn't end up in `ldconfig` / `LD_LIBRARY_PATH` or any of the standard search paths. We hope to address this in the near future.